arch/x86/include/intel64/kernel_arch_func.h

```diff
@@ -8,11 +8,9 @@
 
 #ifndef _ASMLANGUAGE
 
-static ALWAYS_INLINE void
-z_set_thread_return_value(struct k_thread *thread, unsigned int value)
-{
-	thread->callee_saved.rax = value;
-}
+/* FIXME: z_arch_switch() should be declared in a kernel header */
```

Comment:
Remove comment.  kernel_arch_func.h is, in fact, the header responsible for declaring this stuff, we don't have a generic "kernel_arch_api.h" header anywhere (though I don't argue whether or not we should, and in fact we're not terribly consistent about where different architectures declare different functions).  Most architectures require a bunch of these functions to be inline anyway -- it's never in general legal to be compiling in an environment where you see a portable declaration without the arch one.

Reply:
> we don't have a generic "kernel_arch_api.h" header anywhere (though I don't argue whether or not we should, and in fact we're not terribly consistent about where different architectures declare different functions). 

I'm busy working on this as we speak, all the z_arch functions will be in one header (including z_arch_switch())